### DISCUSSION TITLE
Temperature & Humidity Sensors - "Undefined" Output from Tuya API - Help Please

### DISCUSSION THREAD

**Discussion by Tom5640** [milo526/homebridge-tuya-web#278] [Q&A]
Hey All,

Love the plugin and works with my lights and star projector perfectly.  I bought a couple of temperature sensors, and changed the device type as follows (scrambled the IDs with xxxxxx):

![IMG_1180](https://user-images.githubusercontent.com/73896025/119565833-5dbdbf00-bda2-11eb-9d90-d743015fbddc.jpeg)

`[
    {
        "id": "bf0f627xxxxxxx2f7py0x",
        "device_type": "temperature_sensor"
    },
    {
        "device_type": "temperature_sensor",
        "id": "bf55873xxxxxxxx76342wdr"
    }
]`

I receive the following in the log:

`25/05/2021, 21:37:52 [Tuya] [LT Temperature Humidity Sensor] - Requesting device state
25/05/2021, 21:37:52 [Tuya] [LT Temperature Humidity Sensor] - Creating new debounced promise
25/05/2021, 21:37:52 [Tuya] [LT Temperature Humidity Sensor] - Triggering debouncedDeviceStateRequest
25/05/2021, 21:37:52 [Tuya] [LT Temperature Humidity Sensor] - Unsetting debouncedDeviceStateRequestPromise
25/05/2021, 21:37:52 [Tuya] [LT Temperature Humidity Sensor] - Resolving resolveDeviceStateRequest from cache
25/05/2021, 21:37:52 [Tuya] [LT Temperature Humidity Sensor] Characteristic.CurrentTemperature - [GET] undefined`

And the homekit is displayed as per attached:

![IMG_1181](https://user-images.githubusercontent.com/73896025/119565802-54cced80-bda2-11eb-9099-d476c10482fb.jpeg)

I am nearly there with getting this working I think, am I missing something silly?

Thanks,
Tom

**Comment by milo526** [comment]
Please post the result of [the steps under "How to check whether the api this library uses can control your device".](https://github.com/milo526/homebridge-tuya-web#how-to-check-whether-the-api-this-library-uses-can-control-your-device)

Without this it is hard for me to help you with regards to this specific issue.

**Reply by Tom5640** [reply]
No problem. Will do as soon as I am on a computer this evening. Thank you very much.

**Reply by Tom5640** [reply]
> Please post the result of [the steps under "How to check whether the api this library uses can control your device".](https://github.com/milo526/homebridge-tuya-web#how-to-check-whether-the-api-this-library-uses-can-control-your-device)
> 
> Without this it is hard for me to help you with regards to this specific issue.

Getting credentials
Got credentials
{   'access_token': 'EUheu1597451691303bRbj674H0Lv76Em',
    'expires_in': 864000,
    'refresh_token': 'EUheu1597451691303bRbj6hZaT7Sn8gd',
    'token_type': 'bearer'}
Getting devices
Got devices
{   'header': {'code': 'SUCCESS', 'payloadVersion': 1},
    'payload': {   'devices': [   {   'data': {'online': True, 'state': False},
                                      'dev_type': 'switch',
                                      'ha_type': 'switch',
                                      'icon': 'https://images.tuyaeu.com/smart/program_category_icon/wsdcg.png',
                                      'id': 'bf0f6274f4255d62f7py0x',
                                      'name': 'LT Temperature Humidity Sensor'},
                                  {   'data': {   'brightness': '198',
                                                  'color_mode': 'colour',
                                                  'online': True,
                                                  'state': 'false'},
                                      'dev_type': 'light',
                                      'ha_type': 'light',
                                      'icon': 'https://images.tuyaeu.com/smart/icon/ay149682747096840kwW/15934024938d7e111709e.png',
                                      'id': 'bfa0394012be52ef82a8z6',
                                      'name': 'Smart Star Projector'},
                                  {   'data': {   'brightness': '255',
                                                  'color_mode': 'white',
                                                  'color_temp': 5518,
                                                  'online': True,
                                                  'state': 'false'},
                                      'dev_type': 'light',
                                      'ha_type': 'light',
                                      'icon': 'https://images.tuyaeu.com/smart/icon/ay1545812972898u8wAi/1561686999929531d7bc1.jpg',
                                      'id': '77205072d8bfc0f36601',
                                      'name': 'EN Ceiling Light '},
                                  {   'data': {'online': True, 'state': False},
                                      'dev_type': 'switch',
                                      'ha_type': 'switch',
                                      'icon': 'https://images.tuyaeu.com/smart/program_category_icon/wsdcg.png',
                                      'id': 'bf55873a13ecf176342wdr',
                                      'name': 'NS Temperature Humidity Sensor'},
                                  {   'data': {   'brightness': '255',
                                                  'color_mode': 'white',
                                                  'color_temp': 5518,
                                                  'online': False,
                                                  'state': 'true'},
                                      'dev_type': 'light',
                                      'ha_type': 'light',
                                      'icon': 'https://images.tuyaeu.com/smart/icon/ay1545812972898u8wAi/1561686999929531d7bc1.jpg',
                                      'id': '527653045002915e7060',
                                      'name': 'BA Ceiling Light '},
                                  {   'data': {},
                                      'dev_type': 'scene',
                                      'ha_type': 'scene',
                                      'id': 'gau5MMixu7ffUId8',
                                      'name': 'If LT Temperature Humidity '
                                              'Sensor Current Temperature : '
                                              '<10.0°C,Send '
                                              'Notification→Message '}],
                   'scenes': []}}
!!! NOW REMOVE THIS FILE, SO YOUR CREDENTIALS (username, password) WON'T LEAK !!!


**Reply by Tom5640** [reply]
Hey there - did you have a chance to look at the above for me please? Tom

**Reply by milo526** [reply]
Sorry, I forgot about this one. Thanks for pinging me ;) 

```
 { 'data': {'online': True, 'state': False},
'dev_type': 'switch',
'ha_type': 'switch',
'icon': 'https://images.tuyaeu.com/smart/program_category_icon/wsdcg.png',
'id': 'bf0f6274f4255d62f7py0x',
'name': 'LT Temperature Humidity Sensor'},
```

We can see that the data property of this device does not expose a temperature.
Sadly this plugin will not support these devices since we simply cannot get the temperature from them.

Same for the second one.

```
{ 'data': {'online': True, 'state': False},
'dev_type': 'switch',
'ha_type': 'switch',
'icon': 'https://images.tuyaeu.com/smart/program_category_icon/wsdcg.png',
'id': 'bf55873a13ecf176342wdr',
'name': 'NS Temperature Humidity Sensor'}
```

**Reply by Tom5640** [reply]
Ah no problem Milo - my apologies if this was very obvious to anyone with software development experience.  Again, I must thank you very much for the plugin as it works so well with Hoobs for my other Tuya products.  Best wishes, Tom.
